src/main/java/cat/udl/eps/softarch/tfgfinder/domain/Agree.java

```diff
@@ -0,0 +1,45 @@
+package cat.udl.eps.softarch.tfgfinder.domain;
+
+import jakarta.persistence.*;
+import jakarta.validation.constraints.NotBlank;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import org.springframework.format.annotation.DateTimeFormat;
+
+import java.time.ZonedDateTime;
+import java.util.Collection;
+
+@Entity
+@Table(name = "Agrees")
+@Data
+@EqualsAndHashCode(callSuper = true)
+public class Agree extends UriEntity<String> {
+
+    @Id
+    private String id;
+
+    @DateTimeFormat
+    private ZonedDateTime when;
+
+    public enum Status {
+        PENDING,
+        ACCEPTED,
+        REJECTED
+    }
+
+    @Enumerated(EnumType.STRING)
+    private Status status;
+
+    public String getId() { return id; }
```

Comment:
You can remove getters and setters because they are already implemented by using the Lombok annotation @Data